File: Algorithms/Sorting_Algorithms/Radix_Sort/radixsort.cpp
#include<iostream>
using namespace std;

int max(int arr[], int n)
{
	int max = arr[0];
	for (int i = 1; i < n; i++)
		if (arr[i] > max)
			max = arr[i];
	return max;
}


void counting_sort(int arr[], int n, int j)
{
	int output[n]; 
	int i, count[10] = { 0 };

	for (i = 0; i < n; i++)
		count[(arr[i] / j) % 10]++;

	for (i = 1; i < 10; i++)
		count[i] += count[i - 1];

	for (i = n - 1; i >= 0; i--) {
		output[count[(arr[i] / j) % 10] - 1] = arr[i];
		count[(arr[i] /j) % 10]--;
	}

	for (i = 0; i < n; i++)
		arr[i] = output[i];
}


void radix_sort(int arr[], int n)
{
	
	int m = max(arr, n);
	for (int j = 1; m / j > 0; j *= 10)
		counting_sort(arr, n, j);
}

int main()
{
    int n = 10; 
	int arr[] = { 12, 89, 45, 4, 11, 67, 32, 100, 3, 33 };
	radix_sort(arr, n);
	for (int i = 0; i < n; i++)
		cout<<"\t "<<arr[i];
	return 0;
}

File: Algorithms/Sorting-Algorithm/Radix_Sort/radixSort.php

<?php
//Radix Sort algorithm
// Description: The idea of Radix Sort is to divide the array in digit by digit starting from least significant digit to most significant digit. Radix sort uses counting sort as a subroutine to sort.
// Time Complexity: O((n+b) * logb(k))

// A function to do counting sort of arr[] 
// according to the digit represented by exp. 
function countSort(&$arr, $size, $exp) 
{ 
	$output = array_fill(0, $size, 0);  
	$count = array_fill(0, 10, 0); 

	// Store count of occurrences in count[] 
	for ($i = 0; $i < $size; $i++) 
		$count[ ($arr[$i] / $exp) % 10 ]++; 

	// Change count[i] so that count[i] 
	// now contains actual position of 
	// this digit in output[] 
	for ($i = 1; $i < 10; $i++) 
		$count[$i] += $count[$i - 1]; 

	// Build the output array 
	for ($i = $size - 1; $i >= 0; $i--) 
	{ 
		$output[$count[ ($arr[$i] / 
						$exp) % 10 ] - 1] = $arr[$i]; 
		$count[ ($arr[$i] / $exp) % 10 ]--; 
	} 

	// Copy the output array to arr[], so 
	// that arr[] now contains sorted numbers
	// according to current digit 
	for ($i = 0; $i < $size; $i++) 
		$arr[$i] = $output[$i]; 
} 

// The main function to that sorts arr[] 
// of size n using Radix Sort 
function radixsort(&$arr, $size) 
{ 
	
	// Find the maximum number to know
	// number of digits 
	$m = max($arr); 

	// Do counting sort for every digit. Note 
	// that instead of passing digit number, 
	// exp is passed. exp is 10^i where i is 
	// current digit number 
	for ($exp = 1; $m / $exp > 0; $exp *= 10) 
		countSort($arr, $size, $exp); 
} 

// Function to print an array 
function printArray(&$arr,$size) 
{ 
	for ($i = 0; $i < $size; $i++) 
		echo $arr[$i] . " "; 
} 

// Input is given here 

$arr = array(170, 43, 77, 90, 21, 820, 2, 65); 
$size = count($arr); 

// Function Call
radixsort($arr, $size); 
printArray($arr, $size); 

?>